mitodl/open-discussions
# Add support for next param to login prompt

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2390 

#### What's this PR do?

- Fixes the login prompt to add the `next` parameter across the entire site
- Upgrades `react-router` to a newer version to support doing the above without having to pass `location` everywhere

#### How should this be manually tested?
Test this in a few locations:
- Favorite a search result
- Create a post button
- Commenting on a post
- Upvoting a comment or a post
- Generally poke around the site and verify navigation (clicking links, back, forward buttons) works
